toolchain/libvecpf.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author:Harish <harish@linux.vnet.ibm.com>
+#
+
+import os
+import re
+
+from avocado import Test
+from avocado import main
+from avocado.utils import build, git, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Libvecpf(Test):
+
+    """
+    Libvecpf is a "Vector Printf Library"
+    """
+
+    def setUp(self):
+        """
+        Build libvecpf
+
+        Source:
+        http://github.com/Libvecpf/libvecpf.git
+        """
+        self.results = None
+        if not distro.detect().name.lower() == 'ubuntu':
+            self.skip('Upsupported OS %s' % distro.detect().name.lower())
+
+        smm = SoftwareManager()
+        for package in ['gcc', 'make']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.error('%s is needed for the test to be run' % package)
+        git.get_repo("http://github.com/Libvecpf/libvecpf.git",
+                     destination_dir=self.srcdir)
```

Comment:
can you please use asset_fetch it enable cache as well please use expire param in asset_fetch module 

url can be : https://github.com/Libvecpf/libvecpf/archive/master.zip

ex: 
`self.fetch_asset("libvecpf.zip", locations="https://github.com/Libvecpf/libvecpf/archive/master.zip", expire='7d')`